src/Distributor/MostExpensiveFirstDistributor.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusPromotionExtensionsPlugin\Distributor;
+
+use Webmozart\Assert\Assert;
+
+final class MostExpensiveFirstDistributor implements MostExpensiveFirstDistributorInterface
+{
+    public function distribute(array $integers, int $amount): array
+    {
+        Assert::allInteger($integers);
+
+        // Do a first round with only first item to eliminate most of non-applicable items
+        $firstItemWithPromotion = ($integers[0] - $amount);
+        $applicableIntegers = \array_filter($integers, function ($integer) use ($firstItemWithPromotion): bool {
+            return $integer > $firstItemWithPromotion;
+        });
+
+        do {
+            $totalWithPromotion = (\array_sum($applicableIntegers) - $amount) / \count($applicableIntegers);
```

Comment:
Could you import the usage of these functions? You know like `use function array_sum`

Reply:
In PhpStorm, you can click `option + enter` to have a context menu to fix it :)